### ISSUE TITLE
Dart Application objects are not unit-testable

### ISSUE THREAD

**Issue by pylaligand** [domokit/mojo#735]
If I create an Application (from `package:mojo/application.dart`) subclass I can't instantiate it without requiring a full Mojo system backing it - it needs a valid MojoHandle and will try to start handling events when created. This gets in the way of unit-testing, which we should be able to do.

/cc @tonygentilcore

**Comment by zanderso** [comment]
Marking this as a duplicate of https://github.com/domokit/mojo/issues/618. Please re-open if this is a mistake.

**Comment by pylaligand** [comment]
This is a different issue. There's no way to create an Application object to test without starting a chain of events which requires a functional Mojo system.

**Comment by zanderso** [comment]
If you could expand on what it is exactly that you'd like to test about an Application, that could be helpful. At the moment I think it's not clear how to address this problem, or why a specific solution for this problem is needed after #618 is closed.

**Comment by pylaligand** [comment]
The moment you instantiate an Application object, it tries to begin handling events and fails if an invalid Mojo handle is passed. That makes it very hard to create an instance of an Application subclass and unit-test it.

**Comment by zanderso** [comment]
Why can't you make apptests that run in mojo_shell? It seems like this is exactly what they're for.

**Comment by pylaligand** [comment]
I'd like to keep the tests as small and fast as possible, and test only the logic I add to my Application subclass. I shouldn't need to bring up a full Mojo shell for that.

**Comment by zanderso** [comment]
The plan for this situation is to use configuration specific imports to run tests in the standalone VM. No other solution to this problem is currently planned, but if you have a more detailed proposal, I'd certainly be happy to chat about it. Since this is the solution that will be closing #618, why don't we wait until the fix there lands, and see if it addresses the size and performance concerns? Thanks!

**Comment by zanderso** [comment]
It looks like you should also be able to instantiate an Application using an unbound ApplicationStub, which doesn't require any MojoHandle:

https://github.com/domokit/mojo/blob/master/mojo/dart/packages/mojo/lib/src/application.dart#L83

So you'd do something like:

```dart
class MyApplication extends Application {
  MyApplication.standalone() : super.fromStub(new ApplicationStub.unbound());
}
```

**Comment by pylaligand** [comment]
Looks like it works, thanks.
